payjoin/src/core/receive/v2/mod.rs

```diff
@@ -563,7 +563,7 @@ impl Receiver<MaybeInputsOwned> {
     /// An attacker can try to spend the receiver's own inputs. This check prevents that.
     pub fn check_inputs_not_owned(
         self,
-        is_owned: impl Fn(&Script) -> Result<bool, ImplementationError>,
+        is_owned: &mut impl FnMut(&Script) -> Result<bool, ImplementationError>,
```

Comment:
V2 is taking a mutable refrence of the closure but v1 does not. Is there a reason we want this discrepancy?

Reply:
Should we instead go a step further and make even v1 use a reference to the closure?